lib/ccf/one_subject_run_status_per_scan_checker.py

```diff
@@ -7,6 +7,7 @@
 
```

Comment:
So maybe it's possible to combine this class with the `one_subject_run_status_checker` class. ??

This class uses the `extra` attribute of the subject_info as the scan specification that gets included in the `file_name`. So maybe we do the same thing here that we've done elsewhere and include that scan information only if it's been specified?  The problem/question is, "What do we do when that `extra` attribute doesn't specify exactly one scan? I think I named that attribute `extra` instead of just `scan` because sometimes it is specified as `ALL` or `all` instead of as a particular scan name. But we don't want `ALL` or `all` in the run status file name.You should look at the code for places where the `extra` attribute is used and check to see if it can reasonably be changed to be the `scan` attribute instead, and then make sure it makes sense for it to sometimes have the value of `ALL` or `all`. Maybe make the value of `ALL` a constant that is defined in the `subject` class as a special value for all scans. Then in the code in this class you can check to see if the `scan` value is either unset or set to `ALL`. If it is either of those things, then it doesn't get included in the `file_name`.

Depending on your opinion about this, I'd be fine with *not* doing any of this right now (just leave things so there are two classes: `one_subject_run_status_checker` and `one_subject_run_status_per_scan_checker`, and save this for a separate change/feature/review later.